packages/nimble-components/src/wafer-map/specs/features/data-manager-refactor.md

```diff
@@ -0,0 +1,136 @@
+# Nimble Wafer Map ataManager Refactor
+
```

Comment:
Another alternative may be to model the required changes as a state machine using [`@xtate/fsm`](https://xstate.js.org/docs/packages/xstate-fsm/#features).
I think it would be very similar to the UpdateTacker but with a declarative representation instead of procedural in code, i.e. emitting Events at the update tracking machine results in context flags being set.

Reply:
To follow-up here I'll say for the purposes of keyboard state management we found the additional complexity not worth it right now for the table. We could see if it's more useful in the future but I probably wouldn't spend too much time going this direction at the moment